File: baekjoon/greedy/1026_보물.py
"""
1. 아이디어
- A 는 오름차순, B 는 내림차순으로 정렬
- for 문으로 A[index] * B[index] 연산을 한다.

2. 시간복잡도
- for : O(N)

3. 자료구조
- A배열 : int[]
- B배열 : int[]
- 연산결과 cnt : int
"""

import sys
input = sys.stdin.readline

N = int(input())
A = list(map(int, input().split()))
B = list(map(int, input().split()))
cnt = 0

A.sort()
B.sort(reverse=True)

for i in range(N):
    cnt += A[i] * B[i]

print(cnt)

File: baekjoon/greedy/1026_보물.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)
var wr = bufio.NewWriter(os.Stdout)

func scanInt() int {
	sc.Scan()
	text := sc.Text()
	v, _ := strconv.Atoi(text)
	return v
}

func main() {
	defer wr.Flush()
	sc.Split(bufio.ScanWords)

	n := scanInt()
	a, b := make([]int, n), make([]int, n)

	for i := range a {
		a[i] = scanInt()
	}
	for i := range b {
		b[i] = scanInt()
	}

	sort.Ints(a)
	sort.Ints(b)
	sort.Sort(sort.Reverse(sort.IntSlice(b)))

	sum := 0
	for i, value := range a {
		sum += value * b[i]
	}
	fmt.Fprintln(wr, sum)
}
